ote/src/clj/ote/services/external.clj

```diff
@@ -20,25 +33,46 @@
 (defn parse-response->csv
   "Convert given vector to map where map key is given in the first line of csv file."
   [csv-data]
-  (map zipmap
-       (->> (first csv-data) ;; First row is the header
-            (map keyword) ;; Drop if you want string keys instead
-            repeat)
-       (rest csv-data)))
+  (let [headers (first csv-data)
+        valid-header? (valid-csv-header? headers)
+        parsed-data (when valid-header?
+                      (map (fn [[business-id name]]
+                             {::t-service/business-id business-id
+                              ::t-service/name name})
+                            (rest csv-data)))
+        validated-data (filter #(valid-business-id? (::t-service/business-id %)) parsed-data)
+        filtered-data  (map #(dissoc % :remove) validated-data )]
```

Comment:
Why the `#(dissoc % :remove)` I don't see a key named `:remove` in the parsed data, perhaps it is stale

Reply:
yep. good catch, your mapping magic removed the need for :remove key.